lib/langchain/utils/token_length/openai_validator.rb

```diff
@@ -75,6 +75,47 @@ def self.validate_max_tokens!(content, model_name, options = {})
           max_tokens = super(content, model_name, options)
           [options[:max_tokens], max_tokens].reject(&:nil?).min
         end
+
+        # Copied from https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb
```

Comment:
This is just an example from OpenAI. What about the newer models? e.g gpt-3.5-turbo-1106

Reply:
Also the calculation doesn't consider the tools/functions if being used.

Reply:
> Also the calculation doesn't consider the tools/functions if being used.

I would love to have this fixed!